src/widgets/search-box/search-box.js

```diff
@@ -162,66 +142,45 @@ const usage = `Usage:
 searchBox({
   container,
   [ placeholder ],
-  [ cssClasses.{input,poweredBy} ],
-  [ poweredBy=false || poweredBy.{template, cssClasses.{root,link}} ],
-  [ wrapInput ],
+  [ cssClasses.{root, input, reset, submit, loadingIndicator} ],
   [ autofocus ],
-  [ searchOnEnterKeyPressOnly ],
+  [ searchAsYouType = true ],
   [ queryHook ]
-  [ reset=true || reset.{template, cssClasses.{root}} ]
+  [ templates.{reset, submit, loadingIndicator} ]
 })`;
 
 /**
- * @typedef {Object} SearchBoxPoweredByCSSClasses
- * @property  {string|string[]} [root] CSS class to add to the root element.
- * @property  {string|string[]} [link] CSS class to add to the link element.
- */
-
-/**
- * @typedef {Object} SearchBoxPoweredByOption
- * @property {function|string} template Template used for displaying the link. Can accept a function or a Hogan string.
- * @property {SearchBoxPoweredByCSSClasses} [cssClasses] CSS classes added to the powered-by badge.
- */
-
-/**
- * @typedef {Object} SearchBoxResetOption
- * @property {function|string} template Template used for displaying the button. Can accept a function or a Hogan string.
- * @property {{root: string}} [cssClasses] CSS classes added to the reset button.
- */
-
-/**
- * @typedef {Object} SearchBoxLoadingIndicatorOption
- * @property {function|string} template Template used for displaying the button. Can accept a function or a Hogan string.
- * @property {{root: string}} [cssClasses] CSS classes added to the loading-indicator element.
+ * @typedef {Ojbect} SearchBoxTemplates
+ * @property {function|string} submit Template used for displaying the submit. Can accept a function or a Hogan string.
+ * @property {function|string} reset Template used for displaying the button. Can accept a function or a Hogan string.
+ * @property {function|string} loadingIndicator Template used for displaying the button. Can accept a function or a Hogan string.
  */
```

Comment:
`submitIcon` and `resetIcon` are missing

Reply:
Actually those are `submit` and `reset`. Do you think it would be better with the suffix?

Reply:
Oh ok, nevermind then!